lib/editor-tokens.js

```diff
@@ -53,19 +53,23 @@ class EditorTokens {
       return;
     }
 
-    this.awaitingTokens = true;
-
     const grammar = atom.grammars.grammarForScopeName('source.java');
     if (!grammar) {
       // Not loaded. Java grammar possibly not installed. Skip refreshing.
       return;
     }
+    const focusedWindow = remote.BrowserWindow.getFocusedWindow();
+    if (!focusedWindow) {
+      // Lost focus. Skip refreshing.
+      return;
+    }
 
+    this.awaitingTokens = true;
```

Comment:
One thing to note here is I moved `this.awaitingToken = true;` below the consistency checks because if either failed then we'd get stuck with a true value and stop tokenizing until there was a restart.